vertical-pod-autoscaler/e2e/v1/recommender.go

```diff
@@ -140,7 +140,18 @@ var _ = RecommenderE2eDescribe("Checkpoints", func() {
 		_, err := vpaClientSet.AutoscalingV1().VerticalPodAutoscalerCheckpoints(ns).Create(context.TODO(), &checkpoint, metav1.CreateOptions{})
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-		time.Sleep(15 * time.Minute)
+		fmt.Println("Sleeping for up to 15 minutes...")
+
+		maxRetries := 90
+		retryDelay := 10 * time.Second
+		for i := 0; i < maxRetries; i++ {
+			list, err := vpaClientSet.AutoscalingV1().VerticalPodAutoscalerCheckpoints(ns).List(context.TODO(), metav1.ListOptions{})
```

Comment:
Doesn't it set in line 156?
If so we can drop line 156

Reply:
And should we break earlier if the vpa was created within the max retries?

Reply:
> Doesn't it set in line 156? If so we can drop line 156

Yup, but it seems that the `list` and `err` in the loop is only scoped to that loop.

This is another option:
```
		var list *vpa_types.VerticalPodAutoscalerCheckpointList
		
		for i := 0; i < maxRetries; i++ {
			list, err = vpaClientSet.AutoscalingV1().VerticalPodAutoscalerCheckpoints(ns).List(context.TODO(), metav1.ListOptions{})
			if err == nil && len(list.Items) == 0 {
				break
			}
			klog.InfoS("Still waiting...")
			time.Sleep(retryDelay)
		}

		gomega.Expect(err).NotTo(gomega.HaveOccurred())
		gomega.Expect(list.Items).To(gomega.BeEmpty())
```

Honestly, I quite like the way it is now, the first declaration is to test if the loop should exit, the second is to test if the test passes.

However, I'm not strongly opinionated on either option. What do you think?

Reply:
> And should we break earlier if the vpa was created within the max retries?

Not sure what you mean here. This test is looking to see if a VPA Checkpoint is deleted within 15 minutes.

Reply:
> > And should we break earlier if the vpa was created within the max retries?
> 
> Not sure what you mean here. This test is looking to see if a VPA Checkpoint is deleted within 15 minutes.

My mistake. ignore it :)

Reply:
> > Doesn't it set in line 156? If so we can drop line 156
> 
> Yup, but it seems that the `list` and `err` in the loop is only scoped to that loop.
> 
> This is another option:
> 
> ```
> 		var list *vpa_types.VerticalPodAutoscalerCheckpointList
> 		
> 		for i := 0; i < maxRetries; i++ {
> 			list, err = vpaClientSet.AutoscalingV1().VerticalPodAutoscalerCheckpoints(ns).List(context.TODO(), metav1.ListOptions{})
> 			if err == nil && len(list.Items) == 0 {
> 				break
> 			}
> 			klog.InfoS("Still waiting...")
> 			time.Sleep(retryDelay)
> 		}
> 
> 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
> 		gomega.Expect(list.Items).To(gomega.BeEmpty())
> ```
> 
> Honestly, I quite like the way it is now, the first declaration is to test if the loop should exit, the second is to test if the test passes.
> 
> However, I'm not strongly opinionated on either option. What do you think?

I see. so leave it as it is, it's just one more api call so I don't mind

Reply:
> > > Doesn't it set in line 156? If so we can drop line 156
> > 
> > 
> > Yup, but it seems that the `list` and `err` in the loop is only scoped to that loop.
> > This is another option:
> > ```
> > 		var list *vpa_types.VerticalPodAutoscalerCheckpointList
> > 		
> > 		for i := 0; i < maxRetries; i++ {
> > 			list, err = vpaClientSet.AutoscalingV1().VerticalPodAutoscalerCheckpoints(ns).List(context.TODO(), metav1.ListOptions{})
> > 			if err == nil && len(list.Items) == 0 {
> > 				break
> > 			}
> > 			klog.InfoS("Still waiting...")
> > 			time.Sleep(retryDelay)
> > 		}
> > 
> > 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
> > 		gomega.Expect(list.Items).To(gomega.BeEmpty())
> > ```
> > 
> >   
> > Honestly, I quite like the way it is now, the first declaration is to test if the loop should exit, the second is to test if the test passes.
> > However, I'm not strongly opinionated on either option. What do you think?
> 
> I see. so leave it as it is, it's just one more api call so I don't mind

Yup, that's what I thought too.

I'm happy to change if any reviewer has any strong opinions.